ghtt/assignment.py

```diff
@@ -273,9 +275,26 @@ def create_repos(ctx, source, yes, students=None, groups=None):
             has_projects=False,
         )
 
+        # dangerous because we use methods with _
+        default_branch = ghtt.config.get('default-branch', 'master')
+        if default_branch != 'master':
+            g_repo._master_branch = g_repo._makeStringAttribute(default_branch)
+
         click.secho("\n\nGenerating repo {}/{}".format(g_org.html_url, repo.name), fg="green")
 
-        subprocess.check_call(["git", "checkout", "master"], cwd=source)
+        try:
+            subprocess.check_call(["git", "checkout", default_branch], cwd=source)
+        except subprocess.CalledProcessError:
+            if default_branch == 'master' and ghtt.config.get('default-branch', None) is None:
+                # If "master" doesn't work AND you did not specify a default-branch in the config,
+                # automatically try "main" which is most of the time the correct alternative.
+                default_branch = 'main'
+                subprocess.check_call(["git", "checkout", default_branch], cwd=source)
+                # also set "main" as created branch
+                g_repo._master_branch = g_repo._makeStringAttribute(default_branch)
+                click.secho("\n\"master\" branch does not exists -> successfully used fallback to \"main\" branch", fg="green")
+            else:
+                raise
```

Comment:
I think it's better to let the user know there is a misconfiguration, instead of trying to fix the issue, because other actions also rely on `default-branch` being correct.

```suggestion
            click.secho(f"The branch `{default_branch}` does not exist in the source repository. Please specify the correct source branch in `ghtt.yaml` using the `default-branch` keyword.")
            raise
```